src/main/java/com/redhat/devtools/intellij/knative/actions/RefreshAction.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.knative.actions;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.ui.treeStructure.Tree;
+import com.redhat.devtools.intellij.common.actions.StructureTreeAction;
+import com.redhat.devtools.intellij.knative.Constants;
+import com.redhat.devtools.intellij.knative.tree.KnEventingNode;
+import com.redhat.devtools.intellij.knative.tree.KnRevisionNode;
+import com.redhat.devtools.intellij.knative.tree.KnRootNode;
+import com.redhat.devtools.intellij.knative.tree.KnServiceNode;
+import com.redhat.devtools.intellij.knative.tree.KnServingNode;
+import com.redhat.devtools.intellij.knative.tree.KnTreeStructure;
+
+import javax.swing.tree.TreePath;
+
+public class RefreshAction extends StructureTreeAction {
+    public RefreshAction() {
+        super(KnRootNode.class, KnServingNode.class, KnServiceNode.class, KnRevisionNode.class, KnEventingNode.class);
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected) {
+        Tree tree = getTree(anActionEvent);
+        if (tree == null) {
+            return;
+        }
+        KnTreeStructure structure = (KnTreeStructure) tree.getClientProperty(Constants.STRUCTURE_PROPERTY);
+        if (Constants.TOOLBAR_PLACE.equals(anActionEvent.getPlace())) {
+            structure.fireModified(structure.getRootElement());
+        } else {
+            selected = StructureTreeAction.getElement(selected);
+            structure.fireModified(selected);
+        }
+
+    }
+
+    @Override
+    public void update(AnActionEvent e) {
+        if (Constants.TOOLBAR_PLACE.equals(e.getPlace())) {
+            e.getPresentation().setVisible(true);
+            Tree t = getTree(e);
```

Comment:
i'm facing some errors inside the `update` method.
Below a couple of them. They are always the same. Problem with casting

![image](https://user-images.githubusercontent.com/49404737/110919070-22236500-831c-11eb-88b7-489ff0e6ec3f.png)

![image](https://user-images.githubusercontent.com/49404737/110919113-2d769080-831c-11eb-85fe-af3ebe120cff.png)

Reply:
This should be fixed with https://github.com/redhat-developer/intellij-common/pull/45 maybe you need to refresh dependency

Reply:
Ah sorry, yes, my fault then. I'll try it with the new common.

Reply:
I confirm it works fine with the new common. Sorry for the wrong report :+1: